packages/application-shell/TEST_UTILS.md

```diff
@@ -0,0 +1,376 @@
+    }
+  />
+);
+```
+
+This component uses `GetApplicationState` which allows it to access the `applicationState` provided by `ApplicationShell`.
+
+A test which verifies that the authenticated user's first name is rendered by this component can look like this:
+
+```jsx
+import { render } from '@commercetools-frontend/application-shell/test-utils';
+
+describe('FirstName', () => {
+  it('should render the authenticated users first name', () => {
+    const { container, user } = render(<FirstName />);
+    expect(container).toHaveTextContent('Sheldon');
+  });
+});
+```
+
+This test renders the `FirstName` component and then verifies that the name _"Sheldon"_ gets printed. _"Sheldon"_ is the name of our default user in tests. You can find the default test data [here](packages/application-shell/src/test-utils.js).
+
+We can make the test more robust by explicitly declaring the authenticated users first name. This ensures that the test keeps working even when the defaults change, and it also makes it easier to follow.
+
+```jsx
+import { render } from '@commercetools-frontend/application-shell/test-utils';
+
+describe('FirstName', () => {
+  it('should render the authenticated users first name', () => {
+    const { container, user } = render(<FirstName />, {
+      user: {
+        firstName: 'Leonard',
+      },
+    });
+    expect(container).toHaveTextContent('Leonard');
+  });
+});
+```
+
+Here we overwrite the authenticated user's `firstName` for our test. The data we pass in gets merged with the default data.
+
+We can also test the case in which no user is authenticated by passing
+
+```jsx
+import { render } from '@commercetools-frontend/application-shell/test-utils';
+
+describe('FirstName', () => {
+  it('should render the authenticated users first name', () => {
+    const { container, user } = render(<FirstName />, { user: null });
+    expect(container).toHaveTextContent('Anonymous');
+  });
+});
+```
+
+### API
+
+This section describes the methods exported by `@commercetools-frontend/application-shell/test-utils`.
+
+`test-utils` is builds on top of `react-testing-library`, so all `react-testing-library` exports are available. It should not be necessary for you to import `react-testing-library` into your tests. The following section describes the additional exports added on top of `react-testing-library` and the overwritten `render` of `react-testing-library` itself.
+
+#### `render(ui: ReactElement, options: Object)`
+
+| Argument              | Type          | Concern           | Description                                                                                                                                                                                                                                                                                                                                                                                                        |
```

Comment:
The defaults don't matter too much. I found it better to describe them in the  `description` as there are some nuances to them sometimes (like merging instead of overwriting). We can link to the `applicationState` docs though. When I wrote this, I was not yet aware that they exist so I had nowhere to link to.